libtransact/src/contract/archive/mod.rs

```diff
@@ -189,13 +189,30 @@ impl SmartContractArchive {
                 file_path.display()
             ))
         })?;
-        let metadata = metadata.ok_or_else(|| {
+        let metadata: SmartContractMetadata = metadata.ok_or_else(|| {
             Error::new(&format!(
                 "invalid scar file: manifest.yaml not found in {}",
                 file_path.display()
             ))
         })?;
 
+        let version_req = VersionReq::parse(version)?;
```

Comment:
Maybe we can just parse the version at the beginning of this method and pass the `VersionReq` to `find_scar`

Reply:
Addressed, had to change the tests as well to account for this change.